desc/objectives/objective_funs.py

```diff
@@ -454,22 +465,30 @@ def jvp_unscaled(self, v, x):
             The number of vectors given determines the order of derivative taken.
         x : ndarray
             Optimization variables.
+        constants : list
+            Constant parameters passed to sub-objectives.
 
         """
-        if not isinstance(v, tuple):
-            v = (v,)
+        v = v if isinstance(v, (tuple, list)) else (v,)
+
+        compute_unscaled = lambda x: self.compute_unscaled(x, constants)
         if len(v) == 1:
-            return Derivative.compute_jvp(self.compute_unscaled, 0, v[0], x)
+            jvpfun = lambda dx: Derivative.compute_jvp(compute_unscaled, 0, dx, x)
+            return jnp.vectorize(jvpfun, signature="(n)->(k)")(v[0])
```

Comment:
"jvp methods for objectives are now vectorized over the tangents for propagating derivatives" what does this mean? as far as propagating derivatives?

Reply:
This is what allows us to use these methods in `ProximalProjection` and `LinearConstraintProjection`, where they're used instead of calculating full jacobian matrices and then doing a matmul, that's all I meant.